src/base/bittorrent/filepriority.h

```diff
@@ -0,0 +1,43 @@
+/*
+ * Bittorrent Client using Qt and libtorrent.
+ * Copyright (C) 2018  Thomas Piccirello <thomas.piccirello@gmail.com>
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * In addition, as a special exception, the copyright holders give permission to
+ * link this program with the OpenSSL project's "OpenSSL" library (or with
+ * modified versions of it that use the same license as the "OpenSSL" library),
+ * and distribute the linked executables. You must obey the GNU General Public
+ * License in all respects for all of the code used other than "OpenSSL".  If you
+ * modify file(s), you may extend this exception to your version of the file(s),
+ * but you are not obligated to do so. If you do not wish to do so, delete this
+ * exception statement from your version.
+ */
+
+#pragma once
+
+namespace BitTorrent
+{
+    enum class FilePriority : int
+    {
+        Ignored = 0,
+        Normal = 1,
+        High = 6,
+        Maximum = 7,
+        Mixed = -1
+    };
+
+    bool isValidFilePriority(const BitTorrent::FilePriority priority);
```

Comment:
@Piccirello 
Please remove the `const` here (header file) and we can merge.

Reply:
@Chocobo1 Why just the header file? Possibly some C++ specifics I don't know.

Reply:
@glassez how is this example any different than [this](https://github.com/qbittorrent/qBittorrent/pull/9799#discussion_r230601668)? The recommendations on passing by const value seem to be inconsistent.

Reply:
>how is this example any different than this?

This has been mentioned many times (including in conversations with you).
It may make sense in function definition, but it is completely meaningless in the function declaration (moreover, it is misleading there).

Reply:
@Piccirello 
>It may make sense in function definition

The usual advantages of using `const`.

>but it is completely meaningless in the function declaration 

Because it is a local variable, `const` or not is not relevant to the users of the the function, so we leave it out.

Reply:
Thanks for the explainer. I've pushed the change.